communication/mavlink_waypoint_handler.cpp

```diff
@@ -297,7 +297,15 @@ static void waypoint_handler_control_time_out_waypoint_msg(mavlink_waypoint_hand
  *
  * \return  The waypoint in local coordinate frame
  */
-static local_position_t waypoint_handler_set_waypoint_from_frame(waypoint_struct_t* current_waypoint, global_position_t origin);
+static waypoint_local_struct_t waypoint_handler_set_waypoint_from_frame(waypoint_struct_t* current_waypoint, global_position_t origin, dubin_state_t* dubin_state);
+
+/**
+ * \brief   Computes the state machine for the Dubin navigation type
+ *
+ * \param   waypoint_handler        The pointer to the waypoint handler structure
+ * \param   waypoint_next           The next waypoint structure
+ */
+static void waypoint_handler_dubin_state_machine(mavlink_waypoint_handler_t* waypoint_handler, waypoint_local_struct_t* waypoint_next);
```

Comment:
You use the pointer to waypoint_handler only to access:
- waypoint_handler->navigation
- waypoint_handler->state->nav_plan_active
- waypoint_handler->position_estimation

Same as `waypoint_handler_set_waypoint_from_frame`, this does not have much to do with waypoint handler. It looks more like a navigation function.